src/vmm/src/builder.rs

```diff
@@ -628,6 +635,19 @@ pub fn configure_system_for_boot(
     Ok(())
 }
 
+fn attach_boot_timer_device(
+    mmio_device_manager: &mut MMIODeviceManager,
+    request_ts: TimestampUs,
+) -> std::result::Result<(), StartMicrovmError> {
+    use self::StartMicrovmError::*;
+
+    mmio_device_manager
+        .register_new_mmio_boot_timer(request_ts.clone())
+        .map_err(RegisterBootTimerDevice)?;
+
+    Ok(())
+}
```

Comment:
The boot-timer device shouldn't be created by `mmio_device_manager.register_new_mmio_boot_timer()`. That function should take the boot-timer-dev as a param and only register it.

The boot-timer device creation can be done in this function in `builder.rs`

Reply:
Done.